DEVDOCS-5878: [Update] graphql-cart-metafields (#314)
bigcommerce/docs: docs/api-docs/storefront/graphql/graphql-cart-metafields.mdx

**GraphQL Storefront API: Cart Metafields**

> # Working with Cart Metafields

Reviewer: ```suggestion
# Working with cart metafields
```

> Cart metafields allow you to add custom metadata to a cart to display product information. Cart metafields are key-value pairs available in REST Storefront Carts APIs, but you can now use cart metafield operations while in GraphQL.

Reviewer: ```suggestion
Cart metafields let you add custom metadata to a cart to display product information. Cart metafields are key-value pairs. You can work with them using both the GraphQL Storefront API and the REST Storefront API.
```

                    }
                  },
                  {
                    "node": {
                      "sku": "s-1138-ME-RE-RE",
                      "metafields": {
                        "edges": []
                      }
                    }
                  },
                  {
                    "node": {
                      "sku": "s-1138-ME-GR-RE",
                      "metafields": {
                        "edges": []
                      }
                    }
                  }
                ]
              }
            }
          },
          {
            "cursor": "YXJyYXljb25uZWN0aW9uOjI=",
            "node": {
              "metafields": {
                "edges": []
              },
              "variants": {
                "edges": [
                  {
                    "node": {
                      "sku": "",
                      "metafields": {
                        "edges": []
                      }
                    }
                  }
                ]
              }
            }
          }
        ]
      }
    }
  }
}
```
</Tab>
</Tabs>

### Create a cart metafield

<Callout type="info">

> The current platform limit is 250 metafields for a single cart. 

Reviewer: ```suggestion
The platform limit is 250 metafields for a single cart. 
```

</Callout>

<Tabs items={['Request', 'Response']}>
  <Tab>
    

>   ```graphql filename="Example mutation: Create a cart metafield" showLineNumbers copy 

Reviewer: the syntax highlighting makes it look like something is off in this block.

In addition, can you separate the variables being sent into a json object in a third tab? That's how we advise people to construct queries.

Author: @slsriehl I think I fixed the spacing. We don't need another tab for variables because the variables are already in the request.

  mutation createCartMetafield {
 
  cart {
  createCartMetafield(input:{
    cartEntityId: "4a5fd706-beb9-41b5-9ac2-01593b471168"
    data: {
      key: "size"
      value: "small"
    }
  }) {
    metafield {
      id
      entityId
      key
      value
    }
    errors {
      ...on Error {
        message
      }
    }
  }
```
</Tab>
  <Tab>
  ```json filename="Example mutation: Create a cart metafield" showLineNumbers copy
  {
  "data": {
    "cart": {
      "createCartMetafield": {
        "metafield": {
          "id": "TWV0YWZpZWxkczo1Mw==",
          "entityId": 53,
          "key": "size",
          "value": "small"
        },
        "errors": []
      }
    }
  }
}
```
  </Tab>
</Tabs>

### Update a cart metafield
<Tabs items={['Request', 'Response']}>
  <Tab>
  ```graphql filename="Example mutation: Update a cart metafield" showLineNumbers copy 
  mutation updateCartMetafield {
  cart {
    updateCartMetafield(input:{
      cartEntityId: "4a5fd706-beb9-41b5-9ac2-01593b471168"
      metafieldEntityId:53
      data: {
        key: "Size",

> 				value: "medium"

Reviewer: fix spacing.

 {
  "data": {
    "cart": {
      "updateCartMetafield": {
        "metafield": {
          "id": "TWV0YWZpZWxkczo1Mw==",
          "entityId": 53,
          "key": "Size",
          "value": "medium"
        },
        "errors": []
      }
    }
  }
}
```
  </Tab>
</Tabs>

### Delete a cart metafield
<Tabs items={['Request', 'Response']}>
  <Tab>
```graphql filename="Example mutation: Delete a cart metafield" showLineNumbers copy 
 mutation deleteCartMetafield {
 cart {
  deleteCartMetafield(input:{
    cartEntityId: "4a5fd706-beb9-41b5-9ac2-01593b471168"
    metafieldEntityId: 53
  }) {
  errors {
        ... on NotFoundError {
          __typename
          message
        }
        ... on ValidationError {
          __typename
          message
          path
        }
        ... on NotAuthorisedError {
          __typename
        }
      }
    }
  }
}
```
</Tab>
  <Tab>
```json filename="Example mutation: Delete a cart metafield" showLineNumbers copy
{
  "data": {
    "cart": {
      "deleteCartMetafield": {
        "errors": []
      }
    }
  }
}
```
  </Tab>
</Tabs>

## Resources

### GraphQL documentation

> - [GraphQL Storefront API overview](/docs/storefront/graphql)

Reviewer: add the GraphQL Storefront API auth doc here too, as well as the playground and reference pages.

 {
  "data": {
    "cart": {
      "updateCartMetafield": {
        "metafield": {
          "id": "TWV0YWZpZWxkczo1Mw==",
          "entityId": 53,
          "key": "Size",
          "value": "medium"
        },
        "errors": []
      }
    }
  }
}
```
  </Tab>
</Tabs>

### Delete a cart metafield
<Tabs items={['Request', 'Response']}>
  <Tab>
```graphql filename="Example mutation: Delete a cart metafield" showLineNumbers copy 
 mutation deleteCartMetafield {
  cart {
   deleteCartMetafield(input:{
     cartEntityId: "4a5fd706-beb9-41b5-9ac2-01593b471168"
     metafieldEntityId: 53
  }) {
  errors {
        ... on NotFoundError {
          __typename
          message
        }
        ... on ValidationError {
          __typename
          message
          path
        }
        ... on NotAuthorisedError {
          __typename
        }
      }
    }
  }
}
```
</Tab>
  <Tab>
```json filename="Example mutation: Delete a cart metafield" showLineNumbers copy
{
  "data": {
    "cart": {
      "deleteCartMetafield": {
        "errors": []
      }
    }
  }
}
```
  </Tab>
</Tabs>

## Resources

### GraphQL documentation

> - [GraphQL Storefront API overview](/docs/storefront/graphql)

Reviewer: not broken

- [GraphQL Storefront API: Carts and Checkout](/docs/storefront/cart-checkout/guide/graphql-storefront)
- [Authenticating requests to the GraphQL Storefront API](/docs/start/authentication/graphql-storefront)

> - [Authentication and Example Requests: BigCommerce-generated JWTs](/docs/start/authentication#bigcommerce-generated-jwts)

Reviewer: link works
